coins/monero/src/wallet/seed/classic/mod.rs

```diff
@@ -0,0 +1,250 @@
+use std::collections::HashMap;
+
+use crc::{Crc, CRC_32_ISO_HDLC};
+use rand_core::OsRng;
+
+use crate::random_scalar;
+
+use super::{SeedError, LanguageName};
+
+mod languages;
+
+pub const CLASSIC_SEED_LENGTH: usize = 24;
+pub const CLASSIC_SEED_LENGTH_WITH_CHECKSUM: usize = 25;
+const LANGUAGE_WORD_COUNT: usize = 1626;
+
+#[derive(Clone)]
+pub struct Language {
+  word_list: Vec<String>,
+  word_map: HashMap<String, usize>,
+  trimmed_word_map: HashMap<String, usize>,
+  language_name: LanguageName,
+  unique_prefix_length: usize,
+}
+
+impl Language {
+  /// populates the word_map and trimmed_word_maps
+  fn populate_maps(
+    &mut self,
+    allow_short_words: bool,
+    allow_duplicate_prefix: bool,
+  ) -> Result<(), SeedError> {
+    if self.word_list.len() != LANGUAGE_WORD_COUNT {
+      Err(SeedError::InvalidLanguageWordList)?;
+    }
+
+    for (i, word) in self.word_list.iter().enumerate() {
+      self.word_map.insert(word.clone(), i);
+
+      if word.len() < self.unique_prefix_length && !allow_short_words {
+        Err(SeedError::InvalidWordInLanguage)?;
```

Comment:
If we're able to load all languages right now, this error is never hit. Accordingly, we don't need to have this error.

Reply:
These errors are there to check the language words sanity, since anyone can just go to any language words file(e.g. `en.json`) and modify it.  If there is a way to make them completion errors instead of runtime, then yes we could remove them.

Reply:
Considering we're not modifying them, we're grabbing them from Monero, that shouldn't be necessary.